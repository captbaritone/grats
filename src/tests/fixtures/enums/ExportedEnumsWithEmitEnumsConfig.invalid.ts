// {"tsClientEnums": "enums.ts"}

/** @gqlEnum */
export enum Color {
  RED = "red",
  GREEN = "green",
  BLUE = "blue",
}

/** @gqlEnum */
export type Status = "PENDING" | "COMPLETE" | "CANCELLED";

/** @gqlType */
type Query = unknown;

/** @gqlField */
export function favoriteColor(_: Query): Color {
  return Color.RED;
}

/** @gqlField */
export function status(_: Query): Status {
  return "PENDING";
}
