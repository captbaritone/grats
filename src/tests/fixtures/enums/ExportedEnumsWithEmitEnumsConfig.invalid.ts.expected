-----------------
INPUT
----------------- 
// {"EXPERIMENTAL__emitEnums": "enums.ts"}

/** @gqlEnum */
export enum Color {
  RED = "red",
  GREEN = "green",
  BLUE = "blue",
}

/** @gqlEnum */
export type Status = "PENDING" | "COMPLETE" | "CANCELLED";

/** @gqlType */
type Query = unknown;

/** @gqlField */
export function favoriteColor(_: Query): Color {
  return Color.RED;
}

/** @gqlField */
export function status(_: Query): Status {
  return "PENDING";
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/enums/ExportedEnumsWithEmitEnumsConfig.invalid.ts:11:1 - error: Type alias enums are not supported when `EXPERIMENTAL__emitEnums` is configured. Use `enum` declarations instead. For example: `export enum Status { PENDING = "pending" }`.

11 export type Status = "PENDING" | "COMPLETE" | "CANCELLED";
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
